src/test/resources/limetrans/output.fix/metafix-wip.json

```diff
@@ -227,14 +299,46 @@
   "RSWK" : [ ]
 }
 {
-  "RSWK" : [ ]
+  "RSWK" : [ {
+    "control" : "true",
+    "subjectTopicName" : "English Literature"
+  }, {
+    "control" : "true",
+    "subjectTopicName" : "English"
+  }, {
+    "control" : "true",
+    "subjectTopicName" : "Languages & Literatures"
+  }, {
+    "control" : "true",
+    "subjectGenre" : "Fiction",
+    "subjectGeoName" : "South Africa"
+  } ]
 }
 {
-  "RSWK" : [ ]
+  "RSWK" : [ {
+    "control" : "true",
+    "subjectTopicName" : {
+      "1" : "Semiconductors",
+      "2" : "Solid state physics"
+    }
```

Comment:
I tried to control these with the unless construction as substitute for occurence 1.But it seems that these are not working as intended.

Reply:
I've only just now merged metafacture/metafacture-fix#118, so your condition `exists("RSWK[].$last.subjectTopicName")` wouldn't have worked. Should work now.

Reply:
Could you also somehow make use of `$first` instead?

Reply:
I would need to $prepend my control field then.

Reply:
Not sure we're talking about the same thing. I was referring to "occurence 1".

Reply:
okay. i understand now. but I create an string field and the sole existence of that field should be enough to stop further processing of this (possible) list

Reply:
This snippet seems to work:

```
set_array("RSWK[]")
do list(path: "650??", "var": "$i")
  do list(path: "$i.a", "var": "$j")
    unless exists("@rswk.subjectTopicName")
      replace_all("$j", "\\.?$", "")
      copy_field("$j", "@rswk.subjectTopicName")
    end
  end
  move_field("@rswk", "RSWK[]")
end
uniq("RSWK[]")
```

Except:

1. `do list` skips subsequent `datafield`s that have identical `tag`/`ind1`/`ind2` combinations; e.g., `Solid state physics` in:
```xml
<datafield tag="650" ind1=" " ind2="0">
  <subfield code="a">Semiconductors.</subfield>
</datafield>
<datafield tag="650" ind1=" " ind2="0">
  <subfield code="a">Solid state physics.</subfield>
</datafield>
<datafield tag="650" ind1="1" ind2="4">
  <subfield code="a">Semiconductors.</subfield>
  <subfield code="0">https://scigraph.springernature.com/ontologies/product-market-codes/P25150</subfield>
</datafield>
<datafield tag="650" ind1="2" ind2="4">
  <subfield code="a">Solid State Physics.</subfield>
  <subfield code="0">https://scigraph.springernature.com/ontologies/product-market-codes/P25013</subfield>
</datafield>
```
2. `uniq` respects the full hash, not only the `subjectTopicName` (which becomes apparent when `subjectIdentifier` gets added as well).

Reply:
I added the file: https://github.com/hbz/limetrans/blob/6cd42478ed1264f1fd8866b7a69d5bfa6daff093/src/test/resources/limetrans/input/mf_alma.json

To see how fix is "reading" the file (even if we do not see the arrays) But they should be seen by the index.numbers of fields.

Reply:
I've been using a custom function to print the records during processing:

```
hbz.limetrans.function.DebugRecord(id: "001")
hbz.limetrans.function.DebugRecord("prefix", id: "001")
hbz.limetrans.function.DebugRecord("prefix", id: "001", pretty: "true")
````

It's included in the base branch now (9471b49).

Reply:
See #119 for issue 1.